app/controllers/v3/auth/federation_controller.rb

```diff
@@ -1,9 +1,61 @@
 module V3
   module Auth
     class FederationController < ApplicationController
-      def oidc; end
+      TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%L000Z'.freeze
 
-      def voms; end
+      def oidc
+        credentials = ::Auth::Oidc.unified_credentials(ENV.select { |name| name.start_with?('OIDC') })
+        headers['X-Subject-Token'] = ::Tokenator.to_token(credentials.to_hash)
```

Comment:
This will be implemented in next pull request, because we need to set up responders gem